### PR TITLE
Dev interact

### PR DESCRIPTION
**Pull request by Butter-89**
We basically finished the interaction part with animal ability tested

### CODE REVIEW

**Comment by Butter-89** [comment]
Basically there is no conflict as for now

**Comment by JerryZhangJingyi** [comment]
Looks good

File: Source/AhriAndBear/Private/Component/AABSurvivalComponent.cpp [hunk 1]
```diff
@@ -47,3 +47,8 @@ FORCEINLINE void UAABSurvivalComponent::TickStat(float& currentValue, int maxVal
 	currentValue = FMath::Clamp<float>(currentValue + delta, 0, maxValue);
 }
 
+void UAABSurvivalComponent::AddSurvivalData(const FSurvivalData& value)
+{
+		
+}
+
```
Comment (Trevor802) [inline Source/AhriAndBear/Private/Component/AABSurvivalComponent.cpp:61]: @greenstack I added this function to fulfill the EventTrigger's need, please implement it or rename it to the function you created, thx

File: Source/AhriAndBear/Public/Component/AABSurvivalComponent.h [hunk 2]
```diff
@@ -68,4 +68,5 @@ class AHRIANDBEAR_API UAABSurvivalComponent : public UActorComponent
 	void TickStat(float& currentValue, int maxValue, float delta);
 
 	FORCEINLINE FSurvivalData GetSurvivalData() const { return FSurvivalData{ CurrentHealth, CurrentFullness, CurrentThirst, CurrentWarmth }; }
```
Comment (Trevor802) [inline Source/AhriAndBear/Public/Component/AABSurvivalComponent.h:75]: Also, this interface is for the EventTrigger's need, please rename it to your function.

**Review by Trevor802 — CHANGES_REQUESTED** [review]
@greenstack I made some changes in the SurvivalComponent to fulfill EventTrigger's need, please change them to the functions you created.

**Review by Rud156 — APPROVED** [review]

**Comment by Rud156** [comment]
Looks good

File: Source/AhriAndBear/ABAnimalCharacter.cpp [hunk 3]
```diff
@@ -109,9 +188,21 @@ bool AABAnimalCharacter::CanInteract()
...
+	if (bInteracting == true)
+	{
+		return false;
+	}
+	else
+	{
+		return true;
+	}
+}
+
```
Comment (greenstack) [inline Source/AhriAndBear/ABAnimalCharacter.cpp:202]: We can simplify this to just `return !bInteracting;`.

File: Source/AhriAndBear/Private/Component/AABSurvivalComponent.cpp (hunk 1 above)
Comment (greenstack) [reply Source/AhriAndBear/Private/Component/AABSurvivalComponent.cpp:61]: I'll take care of this in the `jn_survival_ui` branch once this pull request is merged due to API changes in that branch.

**Review by greenstack — APPROVED** [review]
I've left a comment on how to simplify a function. The compiler's probably smart enough to do that simplification, but for readability, I would recommend that change. Otherwise, we're good to go.

**Review by Trevor802 — APPROVED** [review]

**Pull request merged**